### PR TITLE
feat: update implementation of QAAccuracy to use Transform-based approach

### PR DESCRIPTION
**Pull request by danielezhu**
*Description of changes:*
Title

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/fmeval/eval_algorithms/qa_accuracy.py [hunk 1]
```diff
@@ -45,29 +36,11 @@
 QUASI_EXACT_MATCH_SCORE = "quasi_exact_match_score"
 PRECISION_OVER_WORDS = "precision_over_words"
 RECALL_OVER_WORDS = "recall_over_words"
+SCORE_NAMES = [F1_SCORE, EXACT_MATCH_SCORE, QUASI_EXACT_MATCH_SCORE, PRECISION_OVER_WORDS, RECALL_OVER_WORDS]
 
 logger = logging.getLogger(__name__)
 
 
-@dataclass(frozen=True)
```
Comment (danielezhu) [inline src/fmeval/eval_algorithms/qa_accuracy.py:52]: This got moved lower in the file

File: src/fmeval/eval_algorithms/qa_accuracy.py [hunk 2]
```diff
@@ -215,6 +188,81 @@ def _quasi_exact_match_score(model_output: str, target_output: str) -> float:
 }
 
 
+def _get_score(
```
Comment (danielezhu) [inline src/fmeval/eval_algorithms/qa_accuracy.py:191]: This is old code that just got moved around in the file.

File: test/unit/eval_algorithms/test_qa_accuracy.py [hunk 3]
```diff
@@ -417,157 +382,24 @@ class TestCaseQAAccuracyEvaluate(NamedTuple):
...
-        """
-        get_dataset.return_value = test_case.input_dataset
-        generate_model_predict_response_for_dataset.return_value = test_case.input_dataset_with_generated_model_output
-        eval_algorithm = QAAccuracy(config)
-        actual_response = eval_algorithm.evaluate(model=None, dataset_config=test_case.dataset_config)
-        assert not generate_model_predict_response_for_dataset.called
-        assert not save_dataset.called
-        assert actual_response == test_case.expected_response
-
-    class TestCaseQAAccuracyEvaluateInvalid(NamedTuple):
```
Comment (danielezhu) [inline test/unit/eval_algorithms/test_qa_accuracy.py:491]: Validation of `evaluate` parameters is now handled by `evaluate_dataset` function, which has its own unit tests. Hence these are redundant.

File: src/fmeval/eval_algorithms/qa_accuracy.py (hunk 2 above)
Comment (malhotra18) [inline src/fmeval/eval_algorithms/qa_accuracy.py:191]: Any particular reason to move this out of the class? I would say we keep it inside the class to encapsulate the logic around using `target_output_delimiter` inside the class only. 

We should be good to move it inside `QAAccuracyScores`. right?

File: src/fmeval/eval_algorithms/qa_accuracy.py [hunk 4]
```diff
@@ -238,145 +286,69 @@ class QAAccuracy(EvalAlgorithmInterface):
...
+        """Compute QA accuracy metrics for a single sample.
+
+        :param target_output: The expected/desired model output.
+        :param model_output: The actual model output.
+        :returns: A list of EvalScore objects, one for each of the QA accuracy metrics.
+        """
+        target_output_key = self.transform.target_output_key
+        model_output_key = self.transform.model_output_key
+        sample = {target_output_key: target_output, model_output_key: model_output}
+        pipeline = TransformPipeline([self.transform])
```
Comment (malhotra18) [inline src/fmeval/eval_algorithms/qa_accuracy.py:307]: it feels this is making `evaluate_sample` more complicated, was this concerned raised earlier? I understand we are too late to discuss on this now, but can you share the conclusion or thread if a discussion on this was done earlier?

**Review by nathanng17 — DISMISSED** [review]

**Review by malhotra18 — CHANGES_REQUESTED** [review]

File: src/fmeval/eval_algorithms/qa_accuracy.py (hunk 4 above)
Comment (danielezhu) [reply src/fmeval/eval_algorithms/qa_accuracy.py:307]: It does make things slightly more complicated, but I think it's best to keep things consistent across all algos, where it's clear that `evaluate_sample` and `evaluate` execute the algo's pipeline. Plus, in terms of raw lines of code, there's hardly any increase.

File: src/fmeval/eval_algorithms/qa_accuracy.py (hunk 2 above)
Comment (danielezhu) [reply src/fmeval/eval_algorithms/qa_accuracy.py:191]: Sure, good point.

File: src/fmeval/eval_algorithms/qa_accuracy.py (hunk 4 above)
Comment (malhotra18) [reply src/fmeval/eval_algorithms/qa_accuracy.py:305]: one of the major tenet for `evaluate_sample` has been to keep it simple and readable. I feel we are moving against that by adding transforms in the method. Yeah, let's capture a backlog SIM for it, we can discuss with the team then and take a call. 

I am in favor of not having so many hops for customer in `evaluate_sample` code.

**Review by nathanng17 — APPROVED** [review]

**Review by malhotra18 — APPROVED** [review]

**Pull request merged**